tracer/src/Datadog.Trace.Tools.Runner/Checks/Windows/RegistryService.cs

```diff
@@ -41,5 +41,35 @@ public string[] GetLocalMachineValueNames(string key)
 
             return registryKey?.GetValue(null) as string;
         }
+
+        public bool GetLocalMachineSubKeyVersion(string key, string displayName, out string? tracerVersion)
+        {
+            tracerVersion = null;
+            var versionFound = false;
+
+            if (!RuntimeInformation.IsOSPlatform(System.Runtime.InteropServices.OSPlatform.Windows))
+            {
+                return versionFound;
+            }
+
+            var registryKey = Registry.LocalMachine.OpenSubKey(key);
+
+            if (registryKey != null)
+            {
+                foreach (var subKeyName in registryKey.GetSubKeyNames())
+                {
+                    var subKey = registryKey.OpenSubKey(subKeyName);
+                    var subKeyDisplayName = subKey?.GetValue("DisplayName");
+
+                    if (subKeyDisplayName?.ToString() == displayName)
+                    {
+                        tracerVersion = subKey?.GetValue("VersionMajor") + "." + subKey?.GetValue("VersionMinor");
+                        versionFound = true;
+                    }
+                }
```

Comment:
If that's too complex then so be it, but as much as possible the RegistryService should just abstract the registry, and actual operations (such as looking for the version) should be implemented in the caller code.

Reply:
Thank you for the comments @kevingosse! I made some updates which can be seen on the https://github.com/DataDog/dd-trace-dotnet/commit/e8b60c35b4371dece29dc5d994301cd3b730ed9b commit, let me know what you think of it.